src/goto-instrument/reachability_slicer.cpp

```diff
@@ -193,11 +195,12 @@ void reachability_slicert::slice(goto_functionst &goto_functions)
     {
       Forall_goto_program_instructions(i_it, f_it->second.body)
       {
-        const cfgt::nodet &e=cfg[cfg.entry_map[i_it]];
+        cfgt::nodet &e = cfg[cfg.entry_map[i_it]];
         if(
           !e.reaches_assertion && !e.reachable_from_assertion &&
           !i_it->is_end_function())
           i_it->make_assumption(false_exprt());
+        e.reaches_assertion = e.reachable_from_assertion = false;
```

Comment:
Comment in-inline about why these are reset here?

Reply:
(It looks like the slicer might be run more than once, in which case perhaps it should be reset before use, rather than upon executing the transformation?)

Reply:
It seems that slicer is not really supposed to be called multiple times; or it would require a much more thorough resetting that just the reachability flags. I changed the code to build a new slicer before every call.